stdlib/public/core/StringComparison.swift

```diff
@@ -247,7 +247,10 @@ private func _findBoundary(
     if idx == 0 { return 0 }
 
     let scalar = _decodeScalar(utf8, startingAt: idx).0
-    if scalar._hasNormalizationBoundaryBefore { return idx }
+
+    if scalar._isNFCStarter {
+      return idx
+    }
```

Comment:
Why this change?

Reply:
There was another change that I did somewhere that prompt this, but it seems tests pass either way with this change. Is there any reason tests wouldn't exercise something here that it still makes sense to revert?